### ISSUE TITLE
Would appreciate some docs for Azure AD

### ISSUE THREAD

**Issue by Jack-Barry** [openid/AppAuth-JS#131]
## Expected Behavior

Would like to see some more clarity in documentation for use in Electron apps, specifically around how to use with Azure AD.

## Describe the problem

I have a registered app in Azure AD. The app is configured to allow all the necessary permissions for logging a user in (as far as I know).

![image](https://user-images.githubusercontent.com/8129489/70020495-6ee19300-1552-11ea-8ab8-c88aab295780.png)

A redirect URI is configured for the app at `http://127.0.0.1:8000`.

I believe everything is configured as needed on the Azure side, but am still getting authentication errors.

#### Steps to reproduce the behavior

I cloned the `appauth-js-electron-sample` app, and adjusted variables

```ts
openIdConnectUrl = 'https://login.microsoftonline.com/MY_TENANT_ID'
clientId = MY_CLIENT_ID
scope = 'openid'
```
When I fire up the app and click the _Sign In_ button, it opens a browser window as expected where I am notified that the registered app is requesting permissions. I click _Accept_ and the browser reads "Close your browser to continue". Closing the browser has no effect on the application (does not bring it into focus).

At this point, if I open the devtools in the app I see an error in the console:

![image](https://user-images.githubusercontent.com/8129489/70020387-1a3e1800-1552-11ea-96bf-6e70af359bd1.png)

I have tried changing the scope to `openid offline_access`, but always get the same error. I'm sure there's something stupid I'm missing here, but I have been wrestling it for a few hours making no progress, so would be willing to write up a doc that would help others who need to solve the same problem - just need some pointers on how to do that.

#### Environment

  * AppAuth-JS version: `1.2.6`
  * AppAuth-JS Environment Node/Electron

**Comment by Jack-Barry** [comment]
Nevermind, got Azure AD working in an Electron without using AppAuth.

**Comment by dazbo** [comment]
Hi @Jack-Barry sorry to ask in that closed ticket, but how did you manage to perform Azure AD authentication without AppAuth ?
I'm currently also trying to perform such authentication, and for now, I'm not able to succeed.

That projet AppAuth looks like the good tool for that, but I have the same issue about the unauthorized (or not found in some case).

**Comment by Jack-Barry** [comment]
@dazbo The code I have for this is closed-source, but the basic idea I used for Electron was this:

- Have a function that opens the user's default browser to the applicable Azure Authorization Code endpoint (we use [`open`](https://github.com/sindresorhus/open#readme) to do this part since our app was running on an older version of Electron, but it looks like you can use Electron's built in tool now: [`shell.openExternal`](https://www.electronjs.org/docs/api/shell#shell))
- When the function is called it also spins up a simple Express server on the client that can handle the OAuth redirect (thus your redirect URI for Azure OAuth is `http://127.0.0.1:{SERVER_PORT_NUMBER}`, as per [RFC8252 7.3](https://tools.ietf.org/html/rfc8252#section-7))
- We set a timeout on the redirect server so that it closes if no requests come in after a certain amount of time, and also trigger the server to close once a request has been handled whether it succeeds or fails
- The redirect server emits signals with an `EventEmitter` that can be picked up on by your app to handle successful login/logout/token refresh, failed login/logout/token refresh, token caching, etc.

I know that's a bit vague, but digging around in [Azure's docs](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow) provides what you need to build the authorization code endpoint, and from there the redirect server passes along data to your app. From there it's just a matter of how you want to handle/cache the authorization code and refresh tokens.

I'd highly recommend reading through [RFC8252 - OAuth 2.0 for Native Apps](https://tools.ietf.org/html/rfc8252) to ensure you follow best practices. AppAuth does this for you, but it's not too hard to follow when rolling your own.

Hope this helps

**Comment by dazbo** [comment]
thanks for the details, it's ok and so much for closed-source.
Many thanks!
